packages/flutter_tools/bin/xcode_backend.dart

```diff
@@ -0,0 +1,466 @@
+    if (scriptOutputStreamFileEnv != null && scriptOutputStreamFileEnv.isNotEmpty) {
+      scriptOutputStreamFile = File(
+        scriptOutputStreamFileEnv,
+      ).openSync(mode: FileMode.write);
+    }
+  }
+
+  final Map<String, String> environment;
+  final List<String> arguments;
+
+  void run() {
+    if (arguments.isEmpty) {
+      // Named entry points were introduced in Flutter v0.0.7.
+      stderr.write(
+          'error: Your Xcode project is incompatible with this version of Flutter. '
+          'Run "rm -rf ios/Runner.xcodeproj" and "flutter create ." to regenerate.\n');
+      exit(-1);
+    }
+
+    final String subCommand = arguments.first;
+    switch (subCommand) {
+      case 'build':
+        buildApp();
+        break;
+      case 'thin':
+        // No-op, thinning is handled during the bundle asset assemble build target.
+        break;
+      case 'embed':
+        embedFlutterFrameworks();
+        break;
+      case 'embed_and_thin':
+        // Thinning is handled during the bundle asset assemble build target, so just embed.
+        embedFlutterFrameworks();
+        break;
+      case 'test_observatory_bonjour_service':
+        // Exposed for integration testing only.
+        addObservatoryBonjourService();
+    }
+  }
+
+  bool existsDir(String path) {
+    final Directory dir = Directory(path);
+    return dir.existsSync();
+  }
+
+  bool existsFile(String path) {
+    final File file = File(path);
+    return file.existsSync();
+  }
+
+  /// Run given command in a synchronous subprocess.
+  ///
+  /// Will throw [Exception] if the exit code is not 0.
+  ProcessResult runSync(
+    String bin,
+    List<String> args, {
+    bool verbose = false,
+    bool allowFail = false,
+    String? workingDirectory,
+  }) {
+    if (verbose) {
+      print('♦ $bin ${args.join(' ')}');
+    }
+    final ProcessResult result = Process.runSync(
+      bin,
+      args,
+      workingDirectory: workingDirectory,
+    );
+    echo(result.stdout as String);
+    if ((result.stderr as String).isNotEmpty) {
+      echoError(result.stderr as String);
+    }
+    if (!allowFail && result.exitCode != 0) {
+      stderr.write('${result.stderr}\n');
+      throw Exception(
+        'Command "$bin ${args.join(' ')}" exited with code ${result.exitCode}',
+      );
+    }
+    return result;
+  }
+
+  /// Log message to stderr.
+  void echoError(String message) {
+    stderr.write('$message\n');
+  }
+
+  /// Log message to stdout.
+  void echo(String message) {
+    print(message);
+  }
+
+  /// Exit the application with the given exit code.
+  ///
+  /// Exists to allow overriding in tests.
+  Never exitApp(int code) {
+    exit(code);
+  }
+
+  /// Return value from environment if it exists, else throw [Exception].
+  String environmentEnsure(String key) {
+    final String? value = environment[key];
+    if (value == null) {
+      throw Exception(
+        'Expected the environment variable "$key" to exist, but it did not!',
```

Comment:
Or something similar.
```suggestion
        'Expected the environment variable "$key" to exist, but it was not found.',
```

Reply:
done.